.circleci/config.yml

```diff
@@ -1,53 +1,171 @@
+            unzip redisgears.linux-bionic-x64.master.zip
+            wget http://redismodules.s3.amazonaws.com/redisgears/snapshots/redisgears-dependencies.linux-bionic-x64.master.tgz
+            mkdir -p /var/opt/redislabs/modules/rg
+            cd /var/opt/redislabs/modules/rg
+            tar -xvf /root/redisgears-dependencies.linux-bionic-x64.master.tgz
+
+  platform-build-steps:
+    parameters:
+      platform:
+        type: string
+    steps:
+      - setup-executor
+      - checkout
+      - setup-automation
       - run:
-          name: download_RedisGears_deps
-          command: cd ..; wget http://redismodules.s3.amazonaws.com/redisgears/snapshots/redisgears-dependencies.linux-bionic-x64.master.tgz; mkdir -p /var/opt/redislabs/modules/rg/; cd /var/opt/redislabs/modules/rg/; tar -xvf /root/redisgears-dependencies.linux-bionic-x64.master.tgz
+          name: Build for platform
+          command: |
+            GEARS_VERSION=master OSNICK=<<parameters.platform>> ./build/reqpacks/build
+      - persist_to_workspace:
+          root: bin/artifacts
+          paths:
+            - '*.zip'
+
+jobs:
+  build_and_test:
+    docker:
+      - image: 'ubuntu:bionic'
+    steps:
+      - setup-executor
       - checkout
+      - setup-automation
       - run:
-          name: install_write_behind_to_RedisGears_virtual_env
-          command: /var/opt/redislabs/modules/rg/python3_99.99.99/bin/python3 setup.py install
+          name: Setup Redis
+          command: ./deps/readies/bin/getredis --version 6
+      - setup-gears
       - run:
-          name: initialize_my_sql
+          name: Setup MySQL
           command: |
             service mysql start
-            mysql -u root --execute "CREATE DATABASE test"
-            mysql -u root --execute "CREATE TABLE test.persons (person_id VARCHAR(100) NOT NULL, first VARCHAR(100) NOT NULL, last VARCHAR(100) NOT NULL, age INT NOT NULL, PRIMARY KEY (person_id));"
-            mysql -u root --execute "INSERT INTO mysql.user (User,Host,authentication_string,ssl_cipher,x509_issuer,x509_subject) VALUES('demouser','localhost',PASSWORD('Password123!'),'','','');"
-            mysql -u root --execute "FLUSH PRIVILEGES;"
-            mysql -u root --execute "GRANT ALL PRIVILEGES ON test.* to demouser@localhost;"
-            mysql -u root --execute "FLUSH PRIVILEGES;"
+            ./sbin/setup-mysql
+      - run:
+          name: Install rgsync
+          command: /var/opt/redislabs/modules/rg/python3_99.99.99/bin/python3 setup.py install
+      - run:
+          name: Tests
+          command: python3 -m RLTest --module ../redisgears.so
+
+  platform-build:
+    parameters:
+      platform:
+        type: string
+    docker:
+      - image: 'debian:buster'
+    steps:
+      - platform-build-steps:
+          platform: <<parameters.platform>>
+ 
+  deploy-artifacts:
+    docker:
+      - image: 'cimg/base:2020.01'
```

Comment:
What is this image?

Reply:
This is a the new base minimal image recently introduced by CircleCI. A kind of Ubuntu. Just trying it out. Note that images from CircleCI often require usage of `sudo` to install stuff, and thus things may break in unexpected ways, so in general I prefer taking a standard image like `debian:buster-slim` and configuring it as an executor (e.g. `setup-executor`). But in this case we only needed awscli, so I gave it a try.